### PR TITLE
fix the stop command

### PR DESCRIPTION
**Pull request by devoncarew**
Fix https://github.com/flutter/tools/issues/152 - handle the case where there are multiple pids returned from `lsof`.

@krisgiesing @abarth

### CODE REVIEW

**Comment by abarth** [comment]
LGTM

Do we need this same fix for Windows?

**Comment by devoncarew** [comment]
It looks like that code iterates through all the returned processes, killing each one in turn.

File: lib/src/device.dart [hunk 1]
```diff
@@ -835,14 +835,19 @@ class AndroidDevice extends Device {
...
         _logging.fine('No process to kill for port $_serverPort');
         return true;
       }
-      // Killing a pid with a shell command from within dart is hard,
-      // so use a library command, but it's still nice to give the
-      // equivalent command when doing verbose logging.
+
+      // If there are multiple returned pids, take the first one.
+      if (pid.contains('\n'))
+        pid = pid.split('\n').first.trim();
```
Comment (krisgiesing) [inline lib/src/device.dart]: Don't we want to kill them all?

File: lib/src/device.dart [hunk 2]
```diff
@@ -835,14 +835,19 @@ class AndroidDevice extends Device {
...
+
+      // Killing a pid with a shell command from within dart is hard, so use a
+      // library command, but it's still nice to give the equivalent command
+      // when doing verbose logging.
       _logging.info('kill $pid');
       Process.killPid(int.parse(pid));
```
Comment (krisgiesing) [inline lib/src/device.dart]: Maybe a try/catch in here or a sanity check before int.parse?

File: lib/src/device.dart (hunk 1 above)
Comment (devoncarew) [inline lib/src/device.dart]: Updated to iterate through the processes. I don't know how we're getting multiple for the same port, but it's clear that we are -

File: lib/src/device.dart (hunk 2 above)
Comment (devoncarew) [inline lib/src/device.dart]: Good point; added.

**Comment by devoncarew** [comment]
Updated -

**Comment by krisgiesing** [comment]
LGTM

**Pull request merged**